nixos/modules/services/misc/sonarr.nix

```diff
@@ -48,26 +82,41 @@ in
       "d '${cfg.dataDir}' 0700 ${cfg.user} ${cfg.group} - -"
     ];
 
-    systemd.services.sonarr = {
-      description = "Sonarr";
-      after = [ "network.target" ];
-      wantedBy = [ "multi-user.target" ];
+    systemd.services.sonarr =
+      let
+        # add empty ApiKey so it can be replaced afterwards
+        configContent =
+          settingsCombined // (lib.optionalAttrs (cfg.apiKeyFile != "") { ApiKey = "#APIKEY#"; });
```

Comment:
I think the module should assert that `cfg.settings`does _not_ contain `ApiKey` due to it being a secret in the nix store.

Reply:
added `assertions`